src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -244,46 +250,39 @@ Visitor::visitTypesafe(Parser::GroupGraphPatternSubContext* ctx) {
   };
 
   if (ctx->triplesBlock()) {
-    ops.emplace_back(GraphPatternOperation::BasicGraphPattern{
-        visitTypesafe(ctx->triplesBlock())});
+    ops.emplace_back(visitTypesafe(ctx->triplesBlock()));
   }
   auto others = visitVector<
       std::pair<variant<GraphPatternOperation, SparqlFilter>,
                 std::optional<GraphPatternOperation::BasicGraphPattern>>>(
       ctx->graphPatternNotTriplesAndMaybeTriples());
-  for (auto [variant, triples] : others) {
-    std::visit(ad_utility::OverloadCallOperator{filter, op}, variant);
-    if (triples.has_value()) {
-      if (holds_alternative<SparqlFilter>(variant) && !ops.empty() &&
-          holds_alternative<GraphPatternOperation::BasicGraphPattern>(
-              ops.back().variant_)) {
-        ad_utility::appendVector(
-            get<GraphPatternOperation::BasicGraphPattern>(ops.back().variant_)
-                ._whereClauseTriples,
-            std::move(triples.value()._whereClauseTriples));
-
-      } else {
-        ops.emplace_back(triples.value());
-      }
+  for (auto [graphPattern, triples] : others) {
+    std::visit(ad_utility::OverloadCallOperator{filter, op}, graphPattern);
+    if (!triples.has_value()) {
+      continue;
+    }
+    if (ops.empty() ||
+        !holds_alternative<GraphPatternOperation::BasicGraphPattern>(
+            ops.back().variant_)) {
+      ops.emplace_back(GraphPatternOperation::BasicGraphPattern{});
     }
+    ad_utility::appendVector(
+        get<GraphPatternOperation::BasicGraphPattern>(ops.back().variant_)
+            ._whereClauseTriples,
+        std::move(triples.value()._whereClauseTriples));
```

Comment:
And implement a member function:
```
BasicGraphPattern::appendTriples(const BasicGraphPattern&)
```

Then this place of code (which is crucial) gets much more readable.